app/recurring_job.go

```diff
@@ -375,7 +425,32 @@ func (job *Job) doSnapshot() (err error) {
 	return nil
 }
 
+func (job *Job) deleteSnapshotRetainReachedOrExceeded(volume *longhornclient.Volume) error {
+	volumeAPI := job.api.Volume
+
+	collection, err := volumeAPI.ActionSnapshotList(volume)
+	if err != nil {
+		return err
+	}
+
+	jobSnapshots := filterSnapshotsWithLabel(collection.Data, types.RecurringJobLabel, job.labels[types.RecurringJobLabel])
+	if len(jobSnapshots) >= job.retain && job.retain > MinRecurringJobRetainNumber {
```

Comment:
IIRC, retain counting won't include the volume head. Then the min retain number should be 0 (snapshot prune will be triggered).

Reply:
Setting `MinRecurringJobRetainNumber` to 1 prevents that all snapshots are gone from creating a new snapshot failed as Phan mentioned here https://github.com/longhorn/longhorn-manager/pull/1708#issuecomment-1441107980